subsys/ipc/ipc_service/lib/Kconfig.icmsg

```diff
@@ -27,15 +27,48 @@ config IPC_SERVICE_ICMSG_SHMEM_ACCESS_TO_MS
 	  Maximum time to wait, in milliseconds, for access to send data with
 	  backends basing on icmsg library. This time should be relatively low.
 
-# The Icmsg library in its simplicity requires the system workqueue to execute
-# at a cooperative priority.
-config SYSTEM_WORKQUEUE_PRIORITY
-	range -256 -1
-
 config IPC_SERVICE_ICMSG_BOND_NOTIFY_REPEAT_TO_MS
 	int "Bond notification timeout in miliseconds"
 	range 1 100
 	default 1
 	help
 	  Time to wait for remote bonding notification before the
 	  notification is repeated.
+
+config IPC_SERVICE_BACKED_ICMSQ_WQ_ENABLE
+	bool "Use dedicated workqueue"
+	default y
+	help
+	  Enable dedicated workqueue thread for the ICMsg backend.
+	  Disabling this configuration will cause the ICMsg backend to
+	  process incoming data through the system workqueue context, and
+	  therefore reduces the RAM footprint of the backend.
+	  Disabling this config may result in deadlocks in certain usage
```

Comment:
Please add a note that user-defined callbacks are called from the selected workqueue, so the user is responsible to obey system workqueue rules when this option is disabled.

Reply:
Added the note.